node/src/blockchain/blockchain_bridge.rs

```diff
@@ -309,33 +309,39 @@ impl BlockchainBridge {
             Ok(Some(mbc)) => mbc,
             _ => u64::MAX,
         };
+        let use_unlimited_block_count_range = u64::MAX == max_block_count;
+        let use_latest_block = u64::MAX == start_block_nbr;
         let end_block = match self
             .blockchain_interface
             .lower_interface()
             .get_block_number()
         {
             Ok(eb) => {
-                if u64::MAX == max_block_count || u64::MAX == start_block_nbr {
+                if use_unlimited_block_count_range || use_latest_block {
                     BlockNumber::Number(eb)
                 } else {
                     BlockNumber::Number(eb.as_u64().min(start_block_nbr + max_block_count).into())
                 }
             }
             Err(e) => {
-                if max_block_count == u64::MAX {
-                    info!(
+                if use_unlimited_block_count_range || use_latest_block {
+                    debug!(
                         self.logger,
                         "Using 'latest' block number instead of a literal number. {:?}", e
                     );
                     BlockNumber::Latest
-                } else if u64::MAX == start_block_nbr {
-                    BlockNumber::Latest
                 } else {
+                    debug!(
+                        self.logger,
+                        "Using '{}' ending block number. {:?}",
+                        start_block_nbr + max_block_count,
+                        e
+                    );
                     BlockNumber::Number((start_block_nbr + max_block_count).into())
                 }
             }
         };
-        let start_block = if u64::MAX == start_block_nbr {
+        let start_block = if use_latest_block {
             end_block
```

Comment:
I found an issue you must look at. Given your new test uses only the mock version of the blockchain interface (in order to call `.retrieve_transactions()`), it didn't have a chance to dive in that function. If you consider the parameters your code from here supplies the function with it would most likely cause a panic soon after. 

Notice that you pick `end_block` also to define the value of `start_block`. Because `end_block` is `BlockNumber::Latest` by this time, `start_block` becomes so too. This goes all aligned with the test where you also do params assertion on that method. 

However, go and look inside there, you will bump soon into:

```
 let fallback_start_block_number = match end_block {
            BlockNumber::Number(eb) => eb.as_u64(),
            _ => {
                if let BlockNumber::Number(start_block_number) = start_block {
                    start_block_number.as_u64() + 1u64
                } else {
                    panic!("start_block of Latest, Earliest, and Pending are not supported");
                }
            }
        };
```
I read that code so that the panic is reached because we supplied both the `start_block` and `end_block` equal to `BlockNumber::Latest`.